Feature 348 dmap reformat

bikegeek opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

-   manually compiled docs and verified no warnings, errors and content is as expected 
-   ran all the METreformat tests for DMAP and verified all tests pass
-   generated simple scatter, line, and contour plots using METplotpy using DMAP output

  verified that current tests are passing as a result of updates:
   -  updated the ECNT tests and data (now using latest ensemble stat output)
   -  fixed any affected tests

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

 For @jprestop:
   - verify all tests pass
   - evaluate documentation for clarity of instructions

For @CPKalb:
   - verify all tests pass
   - reformat DMAP output for generating a line plot 
   - reformat DMAP output for generating a scatter plot
   - generate one or both plot types using the appropriate reformatted data:
    - verify line plot can be generated (limited data, so not much to plot)
    - verify scatter plot can be generated (limited data, not much to plot)

### Reformatting

**Run the code on 'seneca':**
METdataio (feature_348_dmap_reformat)
METcalcpy (develop)
METplotpy (feature_23_scatter) - _not yet merged, refactored for Land Project_

**conda environment used**
 (permissions on conda and env are set up for anyone to use)
` /d1/personal/mwin/miniconda3/condabin/conda activate /d1/personal/mwin/miniconda3/envs/mp_analysis_env`

**setting up PYTHONPATH**

(also sets $DATAIO_BASE, $CALCPY_BASE, and $PLOTPY_BASE env vars)
- cd /d1/projects/NRL_DMAP
- switch to bash shell
- `source setup_env.bash`

**Location:**
/d1/projects/NRL_DMAP/

**Reformat the DMAP linetype data for scatter plot and line plot:**
 
**For scatter plot format**:
`python $DATAIO_BASE/METreformat/write_stat_ascii.py  $DATAIO_BASE/METreformat/reformat_dmap_for_scatter.yaml
`

**File called _reformat_dmap_for_scatter_.data is created in the $DATAIO_BASE/METreformat/output directory**

**For line plot format**:
`**python $DATAIO_BASE/METreformat/write_stat_ascii.py  $DATAIO_BASE/METreformat/reformat_dmap_for_line.yaml**
`

**File called reformat_dmap_for_line.scatter is created in the $DATAIO_BASE/METreformat/output directory**

### Generate line or scatter plot to verify that reformatted files can be used by METplotpy

**line plot:**

`python $PLOTPY_BASE/metplotpy/plots/line/line.py  $PLOTPY_BASE/metplotpy/plots/line/custom_line.yaml`

**Creates the _line_from_dmap.png_ file in $PLOTPY_BASE/metplotpy/plots/line**

**scatter plot:**

`python $PLOTPY_BASE/metplotpy/plots/scatter/scatter.py $PLOTPY_BASE/metplotpy/plots/scatter/dmap_scatter.yaml`

**Creates the _scatter_dmap.png_ file in $PLOTPY_BASE/metplotpy/plots/scatter**

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[NA]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Do these changes introduce new SonarQube findings? **[No]**</br>
If **yes**, please describe:

- [x] Please complete this pull request review by **earliest convenience**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **METdataio-X.Y.Z Development** project for official releases
- [ ] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

bikegeek committed:
Update version

georgemccabe committed:
Per dtcenter/METplus#2336, call custom GHA to trigger METplus use case tests when changes are pushed to main_vX.Y and develop branches

bikegeek committed:
Update keep_all_mpr_col to keep_all_cols now that DMAP linetype is also supported

bikegeek committed:
Added test for reformatting the DMAP (grid_stat) linetype for supporting scatter plots

bikegeek committed:
Test config file for reformatted DMAP data used for scatter plot

bikegeek committed:
Replace keep_all_mpr_cols with keep_all_cols now that DMAP linetype can be used for scatter plots, line plots, and contour plots

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METdataio into feature_348_dmap_reformat

bikegeek committed:
update keep_all_mpr_cols to keep_all_cols

bikegeek committed:
update the comment

bikegeek committed:
Removed debug print statements

bikegeek committed:
Added test for reformatted DMAP output for line and contour plots

bikegeek committed:
Merge branch 'main_v3.0' of https://github.com/dtcenter/METdataio into feature_348_dmap_reformat

bikegeek committed:
DMAP sample data

bikegeek committed:
Add values for DMAP reformatting

bikegeek committed:
Config file for testing reformatting DMAP for line and contour plot support

bikegeek committed:
Example data and reformatted data for stat_name/stat_value/confidence level reformatting

bikegeek committed:
Refactored version of original.  Content needed for matched pair data, TCMPR data, and aggregation statistics data

bikegeek committed:
Added content for ranked data values, by all common stat+linetype-specific columns. Made some edits.

bikegeek committed:
New example data

bikegeek committed:
Clean up formatting and wording, added content for the TCDIAG reformatting

bikegeek committed:
Examples for TCDIAG reformatting

bikegeek committed:
Delete METreformat/test/data/ensemble_stat/ensemble_stat_20100101_120000V.stat

no longer relevant due to additional columns introduced in one of the beta releases

bikegeek committed:
Updated the data to the latest version of ECNT (includes additional columns)

bikegeek committed:
Accidentally removed this input file

bikegeek committed:
Merge branch 'main_v3.0' of https://github.com/dtcenter/METdataio into feature_348_dmap_reformat

bikegeek committed:
Added DMAP support

bikegeek committed:
Updated for new ensemble stat ECNT

bikegeek committed:
DMAP reformatting support

bikegeek committed:
change aggregation status in config file

bikegeek committed:
Update the ECNT/ensemble stat tests with latest data and make testing less reliant on hard-coding.

bikegeek committed:
Add better explanation for the input_stats_aggregated setting

bikegeek committed:
update counts for the ensemble stat data directory data: orank from 1426 to 1921 in new data, phist 18 from 9.

bikegeek committed:
updated values for line_data_phist_bin from 180 to 340 and line_data_rhist_rank from 84 to 102

bikegeek committed:
Delete METreformat/test/data/ensemble_stat/ensemble_stat_20100101_120000V.stat

no longer valid/out of date

bikegeek committed:
fix numbe of line_data_phist_bin to 360

bikegeek committed:
Merge branch 'feature_348_dmap_reformat' of https://github.com/dtcenter/METdataio into feature_348_dmap_reformat

bikegeek committed:
update line_data_rhist_rank

bikegeek committed:
Update ECNT aggregation reformatting test to not use hard-coded values

bikegeek committed:
Remove use of named tuple now that test has been simplified

bikegeek committed:
update the yaml file to use keep_all_cols and updated DMAP line type support in comment

bikegeek committed:
Added reformatting information for ECNT

bikegeek committed:
Example reformatted output for ECNT

bikegeek committed:
Cleaned up format in 6.3.1 for easier reading.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
-MET configuration file).
+input formats.  The formats vary, based on the plot type (e.g. a histogram plot requires ranks and values column data,
+whereas a line plot will need statistic name and statistic value columns). The MET .stat/.tcst ASCII output generated
+by MET tools such as the point-stat, grid-stat, ensemble-stat, and tc-pairs tools may contain ASCII columnar data
```
```suggestion
by MET tools such as the Point-Stat, Grid-Stat, Ensemble-Stat, and TC-Pairs tools may contain ASCII columnar data
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
-In the MET .stat and .tcst files, there are numerous rows of data that have a different number of columns.
+MET .stat and .tcst files can be comprised of numerous rows of data with different number of columns. This is a
+result of requesting multiple linetypes in a MET configuration file (e.g. requesting .
 All MET .stat line types have common columns (refer to table 11.1 in the MET User's Guide) that are labelled
```
```suggestion
All MET .stat line types have common columns (refer to `Table 11.1 <https://met.readthedocs.io/en/main_v12.0/Users_Guide/point-stat.html#id5>`_ in the MET User's Guide) that are labelled
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
 
-In the MET .stat and .tcst files, there are numerous rows of data that have a different number of columns.
+MET .stat and .tcst files can be comprised of numerous rows of data with different number of columns. This is a
+result of requesting multiple linetypes in a MET configuration file (e.g. requesting .
```
@bikegeek This line and the next line currently read:
"This is a result of requesting multiple linetypes in a MET configuration file (e.g. requesting . All MET .stat line types have common columns".  Is that how it should read?  I'm confused by the ". All" part and am wondering if it should be "This is a result of requesting multiple linetypes in a MET configuration file (e.g. requesting all MET .stat line types have common columns" instead?

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
+result of requesting multiple linetypes in a MET configuration file (e.g. requesting .
 All MET .stat line types have common columns (refer to table 11.1 in the MET User's Guide) that are labelled
 (i.e. the columns have headers/names).  The remaining columns
 are unlabelled.  When the stat output from the MET point-stat, grid-stat, ensemble-stat, or tc-pairs tools are
```
```suggestion
are unlabelled.  When the stat output from the MET Point-Stat, Grid-Stat, Ensemble-Stat, or TC-Pairs tools are
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
-input formats.  The MET .stat/.tcst ASCII output generated by MET tools such as the point-stat, grid-stat,
-ensemble-stat, and tc-pairs tools may contain ASCII columnar data from multiple line types (determined by settings in the
-MET configuration file).
+input formats.  The formats vary, based on the plot type (e.g. a histogram plot requires ranks and values column data,
```
```suggestion
input formats.  The formats vary, based on the plot type.  For example, a histogram plot requires ranks and values column data,
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
-ensemble-stat, and tc-pairs tools may contain ASCII columnar data from multiple line types (determined by settings in the
-MET configuration file).
+input formats.  The formats vary, based on the plot type (e.g. a histogram plot requires ranks and values column data,
+whereas a line plot will need statistic name and statistic value columns). The MET .stat/.tcst ASCII output generated
```
```suggestion
whereas a line plot will need statistic name and statistic value columns. The MET .stat/.tcst ASCII output generated
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
+input formats.  The formats vary, based on the plot type (e.g. a histogram plot requires ranks and values column data,
+whereas a line plot will need statistic name and statistic value columns). The MET .stat/.tcst ASCII output generated
+by MET tools such as the point-stat, grid-stat, ensemble-stat, and tc-pairs tools may contain ASCII columnar data
+from multiple line types (determined by settings in the MET configuration file).
```
```suggestion
from multiple line types, as determined by settings in the MET configuration file.
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
 (i.e. the columns have headers/names).  The remaining columns
 are unlabelled.  When the stat output from the MET point-stat, grid-stat, ensemble-stat, or tc-pairs tools are
 reformatted, these unlabelled columns are rearranged into a format appropriate
 for the METplotpy plot of interest (based on the *line_type* setting in the YAML configuration file).
```
```suggestion
for the METplotpy plot of interest, based on the *line_type* setting in the YAML configuration file.
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
-METviewer database. A database query is then performed to filter the data.  The query is
-based on:
+When generating plots in METviewer, the MET .stat input data is first loaded into the
+a database. A database query is then performed to filter the data (based on criteria specified through
```
```suggestion
a database. A database query is then performed to filter the data, based on criteria specified through
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
-based on:
+When generating plots in METviewer, the MET .stat input data is first loaded into the
+a database. A database query is then performed to filter the data (based on criteria specified through
+METviewer's graphical user interface).  The query is based on:
```
```suggestion
METviewer's graphical user interface.  The query is based on:
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -7,33 +7,40 @@ Description
+  is this format that is expected by the METplotpy scripts**.
 
 Users have the option to generate plots from the command line, by-passing METviewer and its database. **However,
 the MET stat-analysis** tool will need to be used for performing any necessary filtering of data (e.g. by any combination of
```
```suggestion
the MET Stat-Analysis** tool will need to be used for performing any necessary filtering of data (e.g. by any combination of
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
 grid-stat, ensemble-stat, or tc-pairs tool must be reformatted before invoking the METplotpy scripts from the
```
```suggestion
Grid-Stat, Ensemble-Stat, or TC-Pairs tools must be reformatted before invoking the METplotpy scripts from the
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: FHO
+
+          **Forecast, Hit, Observation Rates**
```
```suggestion
         **Forecast, Hit, Observation Rates**
```
Alignment is off.  Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+          **Forecast, Hit, Observation Rates**
+
+          From MET tools:
```
```suggestion
         From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: CNT
+
+         **Continuous Statistics**
```
```suggestion
        **Continuous Statistics**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Continuous Statistics**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: CTC
+
+          **Contingency Table Counts**
```
```suggestion
         **Contingency Table Counts**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+          **Contingency Table Counts**
+
+          From MET tools:
```
```suggestion
         From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: CTS
+
+         **Contingency Table Statistics**
```
```suggestion
        **Contingency Table Statistics**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Contingency Table Statistics**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: SL1L2
+
+         **Scalar L1L2 Partial Sums**
```
```suggestion
        **Scalar L1L2 Partial Sums**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Scalar L1L2 Partial Sums**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: VL1L2
+
+         **Vector L1L2 Partial Sums**
```
```suggestion
        **Vector L1L2 Partial Sums**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Vector L1L2 Partial Sums**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: ECNT
+
+         **Ensemble Continuous Statistics**
```
```suggestion
        **Ensemble Continuous Statistics**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Ensemble Continuous Statistics**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Multi-category Contingency Table Statistics**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: MCTS
+
+         **Multi-category Contingency Table Statistics**
```
```suggestion
        **Multi-category Contingency Table Statistics**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: VCNT
+
+         **Vector Continuous Statistics**
```
```suggestion
        **Vector Continuous Statistics**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Vector Continuous Statistics**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: PCT
+
+         **Contingency Table counts for Probabilistic forecasts**
```
```suggestion
        **Contingency Table counts for Probabilistic forecasts**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Contingency Table counts for Probabilistic forecasts**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown::  RHIST
+
+         **Ranked Histogram Counts**
```
```suggestion
        **Ranked Histogram Counts**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Ranked Histogram Counts**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown::  TCDIAG (with TCMPR)
+
+         **Tropcial Cyclone Diagnostics**
```
```suggestion
        **Tropcial Cyclone Diagnostics**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+     .. dropdown::  TCDIAG (with TCMPR)
+
+         **Tropcial Cyclone Diagnostics**
+         **Tropical Cyclone Matched Pairs**
```
```suggestion
        **Tropical Cyclone Matched Pairs**
```
Alignment is off. Attempting to fix.

jprestop replied:
**Tropcial Cyclone Diagnostics** and **Tropical Cyclone Matched Pairs** are on the same line when this is rendered in the documentation here https://metdataio.readthedocs.io/en/feature_348_dmap_reformat/Users_Guide/reformat_stat_data.html.  You may need to have a [line block](https://metplus.readthedocs.io/en/develop/Contributors_Guide/documentation.html#line-blocks).  If it is still problematic, I'm sure @lisagoodrich can help.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+         **Tropcial Cyclone Diagnostics**
+         **Tropical Cyclone Matched Pairs**
+
+         From MET tool:
```
```suggestion
        From MET tool:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: MPR
+
+         **Matched Pair data**
```
```suggestion
        **Matched Pair data**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Matched Pair data**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+     .. dropdown:: DMAP
+
+         **Distance Map Statistics**
```
```suggestion
        **Distance Map Statistics**
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+         **Distance Map Statistics**
+
+         From MET tools:
```
```suggestion
        From MET tools:
```
Alignment is off. Attempting to fix.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+          **Forecast, Hit, Observation Rates**
+
+          From MET tools:
+            - point-stat
```
```suggestion
            - point-stat
```
These references to the MET tools from here and below should either be capitalized if referring to the MET tool (e.g. Point-Stat) or with all lowercase and with an underscore (e.g. point_stat) if referring to the executable.  Please fix up the references as intended and I'll take another look.  Thanks!

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+        Used by the following METplotpy plot types:
+            - TCMPR (Tropical Cyclone Matched Pair) plots:
+
```
This extra line does not show up in RTD:
https://metdataio.readthedocs.io/en/feature_348_dmap_reformat/Users_Guide/reformat_stat_data.html
Please ask @lisagoodrich for assistance if needed.

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+         set of common stat column values) to longer format (where there are multiple rows
+         corresponding to the same common stat column values).
+
+           - The common stat columns are common to all line types for point-stat, grid-stat, and ensemble-stat (model, vx mask, description, fcst lead, etc.)
```
```suggestion
           - The common stat columns are common to all line types for Point-Stat, Grid-Stat, and Ensemble-Stat (model, vx mask, description, fcst lead, etc.)
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+           - The common stat columns are common to all line types for point-stat, grid-stat, and ensemble-stat (model, vx mask, description, fcst lead, etc.)
+             as described in the MET User's Guide:
+
+              - table 11.1 (point-stat)
```
```suggestion
              - `Table 11.1 <https://met.readthedocs.io/en/main_v12.0/Users_Guide/point-stat.html#id5>`_ (Point-Stat)
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+             as described in the MET User's Guide:
+
+              - table 11.1 (point-stat)
+              - table 12.1 (grid-stat)
```
```suggestion
              - `Table 12.1 <https://met.readthedocs.io/en/main_v12.0/Users_Guide/grid-stat.html#id8>`_ (grid-stat)
```

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+
+              - table 11.1 (point-stat)
+              - table 12.1 (grid-stat)
+              - table 13.1 (ensemble-stat)
```
```suggestion
              - `Table 13.1 <https://met.readthedocs.io/en/main_v12.0/Users_Guide/ensemble-stat.html#id1>`_ (ensemble-stat)
```

bikegeek replied:
modified and also used the camel-case version of the tool name, separated by '-'

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+            - multiple rows have duplicate common stat column values, resulting in more rows
 
-- stat_value
+              - note that for the rows containing the first **fy** and **oy** stat_names, the common stat columns of
```
```suggestion
              - note that for the rows containing the first **FY** and **OY** stat_names, the common stat columns of
```
Changing these to upper case to be consistent with the MET documentation (https://met.readthedocs.io/en/latest/Users_Guide/grid-stat.html#id9)

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+         - unlabelled columns in the original .stat file are placed under the linetype-specific column names
 
-    - lower level normal confidence limit
+           - refer to the MET User's Guide for point-stat, grid-stat, ensemble-stat, and tc-pairs  for the column names
```
```suggestion
           - refer to the MET User's Guide for Point-Stat, Grid-Stat, Ensemble-Stat, and TC-Pairs  for the column names
```

bikegeek replied:
suggested changes made during an edit of the file (outside of this interface)

jprestop commented on docs/Users_Guide/reformat_stat_data.rst:
```diff
@@ -46,160 +53,634 @@ Plots that are generated from the command line require a YAML configuration file
+             as described in the MET User's Guide:
 
-The corresponding line types for the above plots are as follows:
+              - table 11.1 (point-stat)
```
```suggestion
              - `Table 11.1 <https://met.readthedocs.io/en/main_v12.0/Users_Guide/point-stat.html#id5>`_ (Point-Stat)
```

bikegeek replied:
suggested change made while editing the file for the other tables.

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

use suggested text

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

use suggested reformatting

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

use suggested formatting

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

use suggested reformatting

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

use suggested format

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

use suggested format/reformatting

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested formatting

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested formatting

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested fix

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested fix

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested fix

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested fix

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested fix

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested fix

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

apply suggested

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

applying suggested change

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/reformat_stat_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update reformat_stat_data.rst

updated MET tool names from lower case to camel case with '-' in between the two words

bikegeek committed:
Update reformat_stat_data.rst

Replace text version of table references to links to the MET User's Guide

bikegeek committed:
Address duplicate explicit target name in links for table 11.1, 12.1, and 13.1

jprestop committed:
Capitalizing MET tool names

jprestop committed:
Per #348, updated spacing to attempt to fix formatting and update capitalization of MET tools

jprestop committed:
Per #348, updated spacing to attempt to fix formatting and update capitalization of MET tools

jprestop committed:
Per #348, removing additional space before literal include directive so it will render correctly in RTD

jprestop committed:
Per #348, removing additional space before literal include directive so it will render correctly in RTD

jprestop committed:
Per #348, adjusting alignment for appropriate alignment in rendering

jprestop committed:
Per #348, modified spacing and capitalization

bikegeek committed:
Fixed some ambiguous text that was meant to be removed or reworded from the previous version of documentation

jprestop reviewed (approved):
@bikegeek Thanks for getting that cleaned up!  I approve this request with regard to the documentation. Thanks for all of your work on this task!

CPKalb commented:
Do you have examples of what the output graphics should look like?  I was able to generate the images but want to make sure I'm getting the expected results.

bikegeek commented:
on seneca

/d1/projects/NRL_DMAP/METplotpy/metplotpy/plots/line/mw_output

and

/d1/projects/NRL_DMAP/METplotpy/metplotpy/plots/scatter/mw_output

On Thu, Dec 12, 2024 at 3:48 PM Christina Kalb ***@***.***>
wrote:

> Do you have examples of what the output graphics should look like? I was
> able to generate the images but want to make sure I'm getting the expected
> results.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdataio/pull/354#issuecomment-2540163700>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4UJHUDLY6DJTJTKIQ3G432FIHDJAVCNFSM6AAAAABTJZNS46VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBQGE3DGNZQGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

CPKalb reviewed (approved):
Ran the reformatter and plots, and was able to successfully generate the expected results.

jprestop commented:
@bikegeek I see this issue was merged into a main_v3.1 branch. Should that have been the main_v3.0 branch and the develop branch?  I'm confused because I think we  shouldn't have a main_v3.1 branch yet.  Maybe I do not fully understand the process?

bikegeek commented:
I merged into the 3_1 branch because this is NOT going into the coordinated
release.

On Fri, Dec 13, 2024 at 9:08 AM Julie Prestopnik ***@***.***>
wrote:

> @bikegeek <https://github.com/bikegeek> I see this issue was merged into
> a main_v3.1 branch. Should that have been the main_v3.0 branch and the
> develop branch? I'm confused because I think we shouldn't have a main_v3.1
> branch yet. Maybe I do not fully understand the process?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdataio/pull/354#issuecomment-2541776417>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4UJHT75YLNI2UHBEFHKYL2FMA6ZAVCNFSM6AAAAABTJZNS46VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBRG43TMNBRG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jprestop commented:
@bikegeek I believe there should only be a develop branch at this time and not a main_v3.1 branch.

According to the [release instructions](https://metplus.readthedocs.io/en/develop/Release_Guide/met_development.html#create-release-reference-branch), main_v3.1 should be created only when we're creating the 3.1.0-rc1 release and not prior to.  These instructions are all new since this last 3.0.0-rc1 release and I know it's a bit confusing.  

I believe we need to merge main_v3.1 into develop and delete main_v3.1 now.

bikegeek commented:
But that will put the DMAP changes into the coordinated release, which is
new code.  I thought we weren't supposed to introduce new code after
release candidates?

On Fri, Dec 13, 2024 at 9:41 AM Julie Prestopnik ***@***.***>
wrote:

> @bikegeek <https://github.com/bikegeek> I believe there should only be a
> develop branch at this time and not a main_v3.1 branch.
>
> According to the release instructions
> <https://metplus.readthedocs.io/en/develop/Release_Guide/met_development.html#create-release-reference-branch>,
> main_v3.1 should be created only when we're creating the 3.1.0-rc1 release
> and not prior to. These instructions are all new since this last 3.0.0-rc1
> release and I know it's a bit confusing.
>
> I believe we need to merge main_v3.1 into develop and delete main_v3.1 now.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdataio/pull/354#issuecomment-2541835897>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4UJHWGPIV7ZHFCAT3WRID2FME2DAVCNFSM6AAAAABTJZNS46VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNBRHAZTKOBZG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jprestop commented:
@bikegeek Thanks for the question.  The **_main_v3.0_** branch is the branch that will become the coordinated release.  The current develop branch is intended for development for v3.1.0.  Anything that you want to go into v3.1.0 should go into **_develop_**.  Anything you want in BOTH the coordinated release AND 3.1.0 should go into both **_main_v3.0_** and **_develop_**.  Hopefully, this feature branch [feature_348_dmap_reformat](https://github.com/dtcenter/METdataio/tree/feature_348_dmap_reformat) was created off of develop.  It sounds like you do not want these changes in the 3.0.0 coordinated release, but into the 3.1.0 coordinated release, so they should be merged into develop.  I'll schedule a quick call for now for you, me, @JohnHalleyGotway, and @georgemccabe to ensure we're all on the same page and work through any questions.  It's a bit confusing having switched our methods in the last rc1 release.
